chapters/functions.tex

```diff
@@ -317,10 +319,8 @@ \section{Pure Modelica Functions}\label{pure-modelica-functions}
 allowed.
 \end{nonnormative}
 
-It is possible to mark a function formal parameter as \lstinline!impure!. Only if
-the function formal parameter is marked \lstinline!impure!, it is allowed to pass an
-\lstinline!impure! function to it. A function having a formal function parameter
-marked \lstinline!impure! must be marked \lstinline!pure! or \lstinline!impure!.
+Only if the function formal parameter is \lstinline!impure!, it is allowed to pass an \lstinline!impure! function to it, due to \cref{function-compatibility-or-function-subtyping-for-functions}.
```

Comment:
Maybe this is one of those places where the plain text formatted _impure_ would work even better than `impure`?  There's a small chance that it could reduce the risk of being interpreted as the `impure` being part of the formal parameter's component declaration…

Reply:
With some rephrasing to be more direct, it could look like this:
```suggestion
By \cref{function-compatibility-or-function-subtyping-for-functions}, it follows that an impure function can only be passed as argument to a function formal parameter of impure type.
```